Storage/rest/hypermedia.h

```diff
@@ -31,39 +31,146 @@ SOFTWARE.
 #include "../../TypeSystem/struct.h"
 #include "../../Blocks/HTTP/api.h"
 
+namespace current {
+namespace storage {
+namespace rest {
+
 CURRENT_STRUCT(HypermediaRESTTopLevel) {
   CURRENT_FIELD(url, std::string);
   CURRENT_FIELD(url_status, std::string);
   CURRENT_FIELD(url_data, (std::map<std::string, std::string>));
   CURRENT_FIELD(up, bool);
+
   CURRENT_CONSTRUCTOR(HypermediaRESTTopLevel)(const std::string& url_prefix = "", bool up = false)
       : url(url_prefix), url_status(url_prefix + "/status"), up(up) {}
 };
 
 CURRENT_STRUCT(HypermediaRESTStatus) {
   CURRENT_FIELD(up, bool);
+
   CURRENT_CONSTRUCTOR(HypermediaRESTStatus)(bool up = false) : up(up) {}
 };
 
-CURRENT_STRUCT(HypermediaRESTContainerResponse) {
+CURRENT_STRUCT(HypermediaRESTError) {
+  using T_DETAILS = std::map<std::string, std::string>;
+  CURRENT_FIELD(error, std::string);
+  CURRENT_FIELD(description, std::string);
+  CURRENT_FIELD(details, Optional<T_DETAILS>);
+
+  CURRENT_CONSTRUCTOR(HypermediaRESTError)(const std::string& error, const std::string& description)
+      : error(error), description(description) {}
+  CURRENT_CONSTRUCTOR(HypermediaRESTError)(
+      const std::string& error, const std::string& description, const T_DETAILS& details)
+      : error(error), description(description), details(details) {}
+};
+
+CURRENT_STRUCT(HypermediaRESTGenericResponse) {
+  CURRENT_FIELD(success, bool, true);
+  CURRENT_FIELD(message, Optional<std::string>);
+  CURRENT_FIELD(errors, Optional<std::vector<HypermediaRESTError>>);
```

Comment:
I'd keep a room for extension inside the error object. I can't imagine how I, as an API client, would manage several errors received in response to a single request. If that request was a batch (several operations at once), then, for each item of the batch, I would expect an optional error, again one error per request.